airflow/ui/src/components/DagVersionSelect.tsx

```diff
@@ -0,0 +1,101 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Field } from "@chakra-ui/react";
+import { useQueryClient } from "@tanstack/react-query";
+import type { OptionsOrGroups, GroupBase, SingleValue } from "chakra-react-select";
+import { AsyncSelect } from "chakra-react-select";
+import { useCallback } from "react";
+import { useSearchParams } from "react-router-dom";
+
+import { UseDagVersionServiceGetDagVersionsKeyFn } from "openapi/queries";
+import { DagVersionService } from "openapi/requests/services.gen";
+import type { DAGVersionCollectionResponse, DagVersionResponse } from "openapi/requests/types.gen";
+import { SearchParamsKeys } from "src/constants/searchParams";
+
+type Option = {
+  label: string;
+  value: string;
+};
+
+const DagVersionSelect = ({
+  dagId,
+  disabled = false,
+}: {
+  readonly dagId: string | undefined;
+  readonly disabled?: boolean;
+}) => {
+  const queryClient = useQueryClient();
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const selectedVersion = searchParams.get(SearchParamsKeys.VERSION_NUMBER);
```

Comment:
I think we need additional logic here.

If a TI is selected, show it's version by default.
If a run is selected, show its latest version by default.
If a dag is selected, leave it undefined and we default to the latest version.

So I am sort of fine with this only existing in the component state instead of searchparams.

Reply:
Yes I thought about that.

Reply:
I'll take a look, thanks

Reply:
I'm not sure here, we're at the very top of the DOM there, TI, Run, (just dag we have), are retrieved in nested components.

If we want to bring that up there will be a lot of prop drilling I'm not sure it's great.

Other solution is to use the parameters in the URL to retrieve the dagId, runId, taskId (if defined) and then leverage react query to fetch the TI, DagDetail, or DagRun to get the associated versions.

Reply:
Yes we should be able to use `  const { dagId = "", runId = "", taskId = "" } = useParams();` and then query for the appropriate object to get the version to show. The query should already be in the react-query cache.

Reply:
It's still a bit involved. But I think a user would get very confused to see a graph that isn't the structure for the dag run or task instance they have selected

Reply:
This is all more complicated. I can be a follow up PR